esrally/driver/runner.py

```diff
@@ -592,6 +592,15 @@ def __repr__(self, *args, **kwargs):
         return "node-stats"
 
 
+def search_type_fallback(es, doc_type, index, body, params):
+    if doc_type and not index:
+        index = "_all"
+    path = "/" + "/".join([index, doc_type, "_search"])
```

Comment:
I think we should just use the following here (we agreeed to use `%` formatting in the hot code path):

```python
path = "/%s/%s/_search" % (index, doc_type)
```

IMHO `os.path.join` only makes sense for file system paths but not for URL paths (Otherwise this would create `/index_name\doc_type\search` on Windows?)

Reply:
Yeah `os.path.join` on a second thought doesn't sound like a good idea and `urllib.path.join` is limited for such matters. +1 on your suggestion.